### PR TITLE
Fixes #820: Fix Repeated top mentions in sidebar info-box

### LINKED ISSUE #820: Repeated top mentions in sidebar info-box
**I'm submitting a ...**
- [X] bug report
- [ ] feature request

**Current behavior:**

There is repetition in sidebar info-box for top mentions.

**Expected behavior:**

It should be fixed.

**Screenshots of the issue:**

### PR DESCRIPTION
**Pull request by simsausaurabh**
**Changes proposed in this pull request**
- Sorted topMentions from largest to smallest.
- Fixed repetition of top mentions.

**Screenshots (if appropriate)** 

**Link to live demo: http://pr-826-fossasia-loklaksearch.surge.sh**

**Closes #820**

### CODE REVIEW

**Comment by simsausaurabh** [comment]
@singhpratyush Thanks, yes that really makes more sense. I have updated the patch please review.

File: src/app/feed/info-box/info-box.component.ts [hunk 1]
```diff
@@ -109,18 +105,29 @@ export class InfoBoxComponent implements OnInit, OnChanges {
...
+					acc[curr] += 1;
+				}
+				return acc;
+			}, []);
+			this.topMentions = Object.keys(stored)
+			.map(key => key.trim())
+			.filter(key => key !== '')
+			.map(key => ([key, stored[key]]))
+			.sort((a, b) => b[1] - a[1]);
+			this.topMentions = this.topMentions.slice(0, 10);
```
Comment (singhpratyush) [inline src/app/feed/info-box/info-box.component.ts]: The slice can still go with the previous chained operation. Also, there is one level of indentation missing.

File: src/app/feed/info-box/info-box.component.ts [hunk 2]
```diff
@@ -32,9 +30,7 @@ export class InfoBoxComponent implements OnInit, OnChanges {
 		responsive: true
 	};
 
-	constructor(
-		private store: Store<fromRoot.State>
-	) { }
+	constructor() { }
```
Comment (singhpratyush) [inline src/app/feed/info-box/info-box.component.ts]: Please avoid useless constructors.

File: src/app/feed/info-box/info-box.component.ts [hunk 3]
```diff
@@ -109,18 +105,29 @@ export class InfoBoxComponent implements OnInit, OnChanges {
 
 	}
 	sortMentions(statistics) {
-		let sortable = [];
+		let stored = [];
```
Comment (singhpratyush) [inline src/app/feed/info-box/info-box.component.ts]: What does the name `stored` signify here?

File: src/app/feed/info-box/info-box.component.ts (hunk 2 above)
Comment (simsausaurabh) [reply src/app/feed/info-box/info-box.component.ts]: Yes, I did not removed earlier.

File: src/app/feed/info-box/info-box.component.ts (hunk 3 above)
Comment (simsausaurabh) [reply src/app/feed/info-box/info-box.component.ts]: ```stored``` is the array storing the output sorted on basis of frequency.

File: src/app/feed/info-box/info-box.component.ts (hunk 1 above)
Comment (simsausaurabh) [reply src/app/feed/info-box/info-box.component.ts]: Yes, I am fixing it.

**Comment by simsausaurabh** [comment]
@singhpratyush Please review.

**Pull request merged**